### DISCUSSION TITLE
ElasticSearch Query Editor - Scripted Metric

### DISCUSSION THREAD

**Discussion by shanielh** [grafana/grafana#58398] [Data Sources]
I'm using a custom scripted metric in my elastic queries in Grafana 2.5, I can see why the "manual editor" has gone in 2.6 : I had to add the right level of aggregation in the "customized" editor and then move to the manual and insert my script (with the right aggregation names ("2", "1", etc')).

I used the manual editor just to replace the last aggregation level with a scripted metric. 

I think I'll manage to add a `scripted metric` aggregation with a button for settings that will allow to insert the four fields : `init_script`, `map_script`, `combine_script` and `reduce_script` inside a dialog.

What do you think?

Also, It might be nice to add a `custom` aggregation that will allow to insert the whole aggregation definition as a block, For aggregations that grafana doesn't support at the moment.

**Comment by 1tft** [comment]
That sounds really good. A section for a custom aggregation enables the user to do "everything" and he is not limited by a "simple" query editor. That seperates a pro tool from the rest (-;

**Comment by aocenas** [comment]
There is feature request to reintroduce the raw query mode https://github.com/grafana/grafana/issues/3163.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This looks like it was handled by #3163. If it wasn't, feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
